packages/block-library/src/image/edit.js

```diff
@@ -58,15 +58,21 @@ const LINK_DESTINATION_CUSTOM = 'custom';
 const ALLOWED_MEDIA_TYPES = [ 'image' ];
 
 export const pickRelevantMediaFiles = ( image ) => {
-	let { caption } = image;
+	let { caption, alt } = image;
+	const { filename } = image;
 
 	if ( typeof caption !== 'object' ) {
 		caption = create( { html: caption } );
 	}
 
+	if ( ! alt ) {
+		alt = __( 'This image has an empty alt attribute, the file name is ' ) + filename;
```

Comment:
This needs to use `sprintf()` (example usage: https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/heading/heading-toolbar.js#L18)

Reply:
Also, a bit pedantic but I think the comma should be a semi-colon, so it should be:

```js
alt = sprintf( __( 'This image has an empty alt attribute; its file name is "%s"' ), filename );
```

Reply:
Cool. Changing. Does `caption` also need this?

Reply:
Nope, `caption` is not content that we provide text for; it doesn't needs to be localised.